src/wp-admin/includes/bookmark.php

```diff
@@ -302,6 +305,9 @@ function wp_update_link( $linkdata ) {
 	$link_id = (int) $linkdata['link_id'];
 
 	$link = get_bookmark( $link_id, ARRAY_A );
+	if ( null === $link ) {
+		return new WP_Error( 'invalid_link', __( 'Invalid link ID.' ) );
+	}
```

Comment:
Code discussion with @jrfnl:

The code assumes it `$links` will be an array. However, `get_bookmark()` could also return `null` which is problematic:

```php
$link_cats = $link['link_category'];
```

and
 
```php
$linkdata                  = array_merge( $link, $linkdata );
```

https://3v4l.org/KPmBt 
https://3v4l.org/4vURI